clang/test/Driver/nvptx-cuda-system-arch.c

```diff
@@ -1,4 +1,3 @@
-// REQUIRES: system-linux
```

Comment:
This is failing on z/OS on line 28 (ARCH_sm70 check).  The actual cc1 line is:
```
clang: warning: CUDA version is newer than the latest partially supported version 12.5 [-Wunknown-cuda-version]
clang: error: cannot determine nvptx64 architecture: No NVIDIA GPU detected in the system; consider passing it via '--offload-arch'; environment variable CLANG_TOOLCHAIN_PROGRAM_TIMEOUT specifies the tool timeout (integer secs, <=0 is infinite)
 "/plex/perry/llvm/Woz/build/bin/clang-20" "-cc1" "-triple" "nvptx64-nvidia-cuda" "-aux-triple" "x86_64-unknown-linux-gnu" "-S" "-dumpdir" "a-" "-disable-free" "-clear-ast-before-backend" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "nvptx-cuda-system-arch.c" "-mrelocation-model" "static" "-mframe-pointer=all" "-fno-rounding-math" "-no-integrated-as" "-aux-target-cpu" "x86-64" "-fcuda-is-device" "-mllvm" "-enable-memcpyopt-without-libcalls" "-fcuda-allow-variadic-functions" "-target-cpu" "sm_52" "-target-feature" "+ptx85" "-debugger-tuning=gdb" "-fno-dwarf-directory-asm" "-fdebug-compilation-dir=/plex/perry/llvm/Woz/llvm-project/clang/test/Driver" "-resource-dir" "/plex/perry/llvm/Woz/build/lib/clang/20" "-internal-isystem" "/plex/perry/llvm/Woz/build/lib/clang/20/include/cuda_wrappers" "-include" "__clang_cuda_runtime_wrapper.h" "-internal-isystem" "/plex/perry/llvm/Woz/build/lib/clang/20/include" "-internal-isystem" "/usr/local/include" "-internal-externc-isystem" "/include" "-internal-externc-isystem" "/usr/include" "-internal-isystem" "/plex/perry/llvm/Woz/llvm-project/clang/test/Driver/Inputs/CUDA_102/usr/local/cuda/include" "-internal-isystem" "/plex/perry/llvm/Woz/build/lib/clang/20/include" "-internal-isystem" "/usr/local/include" "-internal-externc-isystem" "/include" "-internal-externc-isystem" "/usr/include" "-fdeprecated-macro" "-fno-autolink" "-ferror-limit" "19" "-nogpulib" "-fgnuc-version=4.2.1" "-mno-csect" "-fskip-odr-check-in-gmf" "-fcxx-exceptions" "-fexceptions" "-fexec-charset" "UTF-8" "-fcolor-diagnostics" "-cuid=15151f95f8ae4b17" "-D__GCC_HAVE_DWARF2_CFI_ASM=1" "-o" "/plex/perry/tmp/nvptx-cuda-system-arch-sm_52-77f5c9.s" "-x" "cuda" "/plex/perry/llvm/Woz/llvm-project/clang/test/Driver/nvptx-cuda-system-arch.c"
```
This has sm_52 as the target_gpu instead of sm_70.  Can you either fix it or add:
```
UNSUPPORTED: system-zos
```

Reply:
The result should definitely be sm_70, that's what the provided script produces

Reply:
I agree.  By fixing it, I was referring to the code in clang that determines this value.  If it's easy to make the test pass on z/OS great.  If not marking it as unsupported works too.  This applies to my other comment too.

Reply:
It's the output from a shell script in clang/test/Driver/Inputs/nvptx-arch/nvptx_arch_sm_70

How could the zos test find a different result

Reply:
I'm looking into that question.  The first thing I noticed was the temp file is being created and read as a binary file.  I also noticed that the output of the script is not being redirected into the temp file.  I'll get a PR up to fix these problems once I get it solved.

I think that mean the only issue on z/OS is not related to these tests.  Could you mark the 3 failing tests as XFAIL for system-zos.  I'll remove that once I have the problem fixed.